pydis_core/exts/source.py

```diff
@@ -0,0 +1,166 @@
+import enum
+import inspect
+from pathlib import Path
+from typing import NamedTuple, TYPE_CHECKING
+
+from discord import Embed
+from discord.ext import commands
+from discord.utils import escape_markdown
+
+if TYPE_CHECKING:
+    from pydis_core import BotBase as Bot
+
+
+class TagIdentifierStub(NamedTuple):
+    """A minmally functioning stub representing a tag identifier."""
+
+    group: str | None
+    name: str
+
+    @classmethod
+    def from_string(cls, string: str) -> "TagIdentifierStub":
+        """Create a TagIdentifierStub from a string."""
+        split_string = string.split(" ", maxsplit=2)
+        if len(split_string) == 1:
+            return cls(None, split_string[0])
+        return cls(split_string[0], split_string[1])
+
+
+class SourceType(enum.StrEnum):
+    """The types of source objects recognized by the source command."""
+
+    help_command = enum.auto()
+    command = enum.auto()
```

Comment:
Lets be explicit.
```suggestion
    prefix_command = enum.auto()
```
This requires changes elsewhere too.

Reply:
I don't really understand, what?

Reply:
Slash commands are not currently supported. Specifying this is a prefix command makes that more clear. If we want to add support for slash commands, this would want to be renamed anyhow for clarity.

Reply:
Oh right, then we should use text_command.